From: Michel Normand <normand@linux.vnet.ibm.com>
Subject: john-1.7.9-powerpc BE need rhash u32 swap copy
Date: Mon, 15 Feb 2016 17:56:13 +0100

rhash_u32_swap_copy needs to be defined for CPU_BIG_ENDIAN
patch required when  1.7.9 jumbo patch is applied
as suggested in uptream ML
http://www.openwall.com/lists/john-dev/2013/02/06/2

Signed-off-by: Michel Normand <normand@linux.vnet.ibm.com>
---
 john-1.7.9-jumbo-7/src/gost_plug.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: john-1.8.0/john-1.7.9-jumbo-7/src/gost_plug.c
===================================================================
--- john-1.8.0.orig/john-1.7.9-jumbo-7/src/gost_plug.c
+++ john-1.8.0/john-1.7.9-jumbo-7/src/gost_plug.c
@@ -531,7 +531,7 @@ int main()
 }
 #endif
 
-#ifndef __GLIBC__
+#ifdef CPU_BIG_ENDIAN
 void rhash_u32_swap_copy(void* to, int index, const void* from, size_t length) {
 	size_t i;
 	unsigned int *pO, *pI;
